client/tasks/task-list.tsx

```diff
@@ -0,0 +1,174 @@
+/**
+ * External dependencies
+ */
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { useEffect, useRef, useState } from '@wordpress/element';
+import { Card, CardBody, CardHeader } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { Badge } from '@woocommerce/components';
+import { ONBOARDING_STORE_NAME } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+import { Text, List, CollapsibleList } from '@woocommerce/experimental';
+
+/**
+ * Internal dependencies
+ */
+import { TaskListItem } from './task-list-item';
+import { TaskListMenu } from './task-list-menu';
+import './task-list.scss';
+
+export const getEventPrefix = ( id: string ): string => {
+	// This helps retain backwards compatibility with the old event naming.
+	if ( id === 'setup' ) {
+		return 'tasklist_';
+	}
+
+	return `${ id }_tasklist_`;
+};
+
+export type TaskProps = {
+	actionLabel?: string;
+	actionUrl?: string;
+	content: string;
+	id: string;
+	isComplete: boolean;
+	isDismissable: boolean;
+	isDismissed: boolean;
+	isVisible: boolean;
+	isSnoozable: boolean;
+	snoozedUntil: number;
+	time: string;
+	title: string;
+};
+
+export type TaskListProps = {
+	id: string;
+	isCollapsible?: boolean;
+	isComplete: boolean;
+	isExpandable?: boolean;
+	tasks: TaskProps[];
+	title: string;
+	query: {
+		task: string;
+	};
+};
+
+export const TaskList: React.FC< TaskListProps > = ( {
+	id,
+	tasks,
+	title: listTitle,
+	isCollapsible = false,
+	isExpandable = false,
+	query,
+} ) => {
+	const { hideTaskList } = useDispatch( ONBOARDING_STORE_NAME );
+	const { profileItems } = useSelect( ( select ) => {
+		const { getProfileItems } = select( ONBOARDING_STORE_NAME );
+
+		return {
+			profileItems: getProfileItems(),
+		};
+	} );
+	const eventPrefix = getEventPrefix( id );
+	const prevQueryRef = useRef( query );
+	const nowTimestamp = Date.now();
+	const visibleTasks = tasks.filter(
+		( task ) =>
+			task.isVisible &&
```

Comment:
Curious, is there a reason non-visible tasks need to be returned from the API at all?

Reply:
I think we can forego them like we did with the other APIs.  Seems like it simplifies things.

Reply:
There's some other logic we could clean up here possibly around dismissed, snoozed, and not visible tasks on the REST side, but I think some changes may be required in the data store to make this function as expected.

I'll open a follow-up issue and address there if you think that's okay.

Reply:
Works for me 👍🏻